x-pack/plugins/global_search_providers/public/providers/get_app_results.ts

```diff
@@ -40,9 +51,17 @@ export const getAppResults = (
 export const scoreApp = (term: string, appLink: AppLink): number => {
   term = term.toLowerCase();
   const title = [appLink.app.title, ...appLink.subLinkTitles].join(' ').toLowerCase();
+  const appScoreByTerms = scoreAppByTerms(term, title);
+  const appKeywords = appLink.app.meta.keywords.map((keyword) => keyword.toLowerCase());
+  const appSubLinkKeywords = appLink.subLinkKeywords.map((keyword) => keyword.toLowerCase());
```

Comment:
If we moved `App.searchDeepLinks` into `App.meta.searchDeepLinks` this would be simpler since we could rollup these into a single field in the `AppLink` type. Should we just do that now?

Reply:
@joshdover should I go ahead and move the `App.searchDeepLinks` into `App.meta.searchDeepLinks` in this PR? If so, I'll need to allocate a bit of time towards it before digging into other work. Since it's planning week anyway, I can try to get it wrapped up before the end of the year.

Reply:
@joshdover how would we declare keywords for deep links if we moved searchDeepLinks into the meta field.
e.g.

before:
```typescript
// keywords in meta:
core.application.register({
  id: 'my_app',
  title: 'Translated title',
  meta: {
    keywords: ['translated keyword1', 'translated keyword2'],
  },
  mount: () => { ... }
})

// searchDeepLinks:
core.application.register({
  id: 'my_app',
  title: 'My App',
  searchDeepLinks: [
    { id: 'sub1', title: 'Sub1', path: '/sub1', meta: { keywords: ['subpath1'] } },
    {
      id: 'sub2',
      title: 'Sub2',
      searchDeepLinks: [
        { id: 'subsub', title: 'SubSub', path: '/sub2/sub', meta: { keywords: ['subpath2'] } }
      ]
    }
  ],
  mount: () => { ... },
})

```

after
```typescript
core.application.register({
  id: 'my_app',
  title: 'My App',
  meta: {
    keywords: ['keyword1', 'keyword2'],
    searchDeepLinks: [
    { id: 'sub1', title: 'Sub1', path: '/sub1' }, // how do we declare keywords for 'Sub1'?
    {
      id: 'sub2',
      title: 'Sub2',
      searchDeepLinks: [
        { id: 'subsub', title: 'SubSub', path: '/sub2/sub' } // how do we declare keywords for 'SubSud'?
      ]
    }
  ],
  }
  
  mount: () => { ... },
})
```

Would this look something like this:

```typescript
core.application.register({
  id: 'my_app',
  title: 'My App',
  meta: {
    keywords: ['keyword1', 'keyword2'], // keywords for 'My App'
    searchDeepLinks: [
    { id: 'sub1', title: 'Sub1', path: '/sub1', keywords: ['Keyword Sub1'] },
    {
      id: 'sub2',
      title: 'Sub2',
      searchDeepLinks: [
        { id: 'subsub', title: 'SubSub', path: '/sub2/sub', keywords: ['Keyword SubSub'] }
      ]
    }
  ],
  }
  
  mount: () => { ... },
})
```

with a slight change in the `AppSearchDeepLink` type:

```typescript
export type AppSearchDeepLink = {
  /** Identifier to represent this sublink, should be unique for this application */
  id: string;
  /** Title to label represent this deep link */
  title: string;
  /** Optional keywords to match with in deep links search */
  keywords?: string[];
} & (
  | {
      /** URL path to access this link, relative to the application's appRoute. */
      path: string;
      /** Optional array of links that are 'underneath' this section in the hierarchy */
      searchDeepLinks?: AppSearchDeepLink[];
    }
  | {
      /** Optional path to access this section. Omit if this part of the hierarchy does not have a page URL. */
      path?: string;
      /** Array links that are 'underneath' this section in this hierarchy. */
      searchDeepLinks: AppSearchDeepLink[];
    }
);
```

Reply:
commit [#e33f299](https://github.com/elastic/kibana/pull/85686/commits/e33f29999fa51a3d07894349ebf53f8af805d9c7) moves `App.searchDeepLinks` into `App.meta.searchDeepLinks` and combines `appKeywords` and `appSublinkKeywords` into a single `keywords` field.

The other refactorings that needed to happen to accomodate the change were also handled in that commit.